src/app/container/register-tool/register-tool.service.ts

```diff
@@ -59,11 +59,23 @@ export class RegisterToolService {
         this.showCustomDockerRegistryPath.next(newShowCustomDockerRegistoryPath);
     }
 
-    isInvalidPrivateTool() {
-        return false;
+    isInvalidPrivateTool(toolObj: Tool) {
+        return toolObj.private_access === true && !toolObj.tool_maintainer_email;
     }
-    isInvalidCustomRegistry() {
-        return false;
+    isInvalidCustomRegistry(toolObj: Tool, customDockerRegistryPath: string) {
+            for (let i = 0; i < this.dockerRegistryMap.length; i++) {
```

Comment:
If the registry is private, it checks if the customDockerRegistryPath is truthy (not null, not undefined, not "", etc)